Optimizer.Launcher/Program.cs

```diff
@@ -63,7 +68,11 @@ public static void Main()
                 Log.Error(e);
             }
 
-            Console.ReadKey();
+            // Wait until the optimizer has stopped running before exiting
+            while (optimizer != null && (optimizer.Status == OptimizationStatus.New || optimizer.Status == OptimizationStatus.Running))
+            {
+                Thread.Sleep(100);
```

Comment:
I think we could directly create a reset event and wait on it and set it in the `optimizer.Ended` callback after disposing of the `optimizer`? So we directly don't need to poll or expose the state

Reply:
Didn't know C# had such a feature, fixed :+1: